Sources/ArgumentParser/Parsing/Interactive.swift

```diff
@@ -56,26 +56,44 @@ extension CommandParser {
 
       // Retrieve the correct `ArgumentDefinition` for the required transformation
       // before storing the new value received from the user.
-      guard
-        let definition = arguments.content.first(where: {
-          $0.help.keys.contains { $0.rawValue == label }
-        })
-      else { break }
-
-      guard case let .unary(update) = definition.update else { break }
-      let name = definition.names.first
-      let updateBy: (String) throws -> Void = { string in
-        try update(InputOrigin(elements: [.interactive]), name, string, &values)
+      let args = arguments.filter { $0.help.keys.contains(key) }
+      let possibilities: [String] = args.compactMap {
+        $0.help.visibility.base == .default
+          ? $0.nonOptional.synopsis
+          : nil
       }
 
-      // All possible strings that can be converted to value
-      // of this CaseIterable enum type.
-      let allValues = definition.help.allValues
-      if allValues.isEmpty {
-        storeNormalValues(label: label, updateBy: updateBy, arguments: arguments)
+      if possibilities.count == 1 {
+        // Missing expected argument
+        let definition = args.first!
+        guard case let .unary(update) = definition.update else { break }
+        let name = definition.names.first
+        let updateBy: (String) throws -> Void = { string in
+          try update(InputOrigin(elements: [.interactive]), name, string, &values)
+        }
+
+        // All possible strings that can be converted to value
+        // of this CaseIterable enum type.
+        let allValues = definition.help.allValues
+        if allValues.isEmpty {
+          storeNormalValues(label: label, updateBy: updateBy, arguments: arguments)
+        } else {
+          allValues.enumerated().forEach { print("\($0 + 1). \($1)") }
+          choose("? Please select '\(label)': ", choices: allValues)
```

Comment:
Since we're passing the list of choices into `choose`, shouldn't `choose` also do the printing of options?